### PR TITLE
refactor: use MdxP element

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Creates a new `MdxP` component, which replaces `.p` styles that had to be independently applied to our `Text` component for docs and tutorial content.

## 🤷 Why

To consolidate MDX content styling across docs and tutorial content into a single place: `components/dev-dot-content/mdx-components`.

## 🛠️ How

- Moves `dev-dot-content` `.p` styles into a new `MdxP` component in `components/dev-dot-content/mdx-components`
- Uses the `MdxP` component for docs and tutorials

## 🧪 Testing

- Visit a docs MDX page, such as [/vault/docs/get-started/developer-qs][/vault/docs/get-started/developer-qs], and confirm headings look as expected (vs [upstream][UPSTREAM/vault/docs/get-started/developer-qs])
- Visit a tutorial MDX page, such as [/waypoint/tutorials/get-started-docker/get-started-ui][/waypoint/tutorials/get-started-docker/get-started-ui], and confirm headings look as expected (vs [upstream][UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui])

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1202097197789424/1202072894754378/f
[preview]: https://dev-portal-git-zsmdx-p-hashicorp.vercel.app/
[/vault/docs/get-started/developer-qs]: https://dev-portal-git-zsmdx-p-hashicorp.vercel.app/vault/docs/get-started/developer-qs
[UPSTREAM/vault/docs/get-started/developer-qs]: https://developer.hashicorp.com/vault/docs/get-started/developer-qs
[/waypoint/tutorials/get-started-docker/get-started-ui]: https://dev-portal-git-zsmdx-p-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-ui
[UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui]: https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-ui

### CODE REVIEW

File: src/components/dev-dot-content/dev-dot-content.module.css [hunk 1]
```diff
@@ -7,93 +7,88 @@
 ***
 */
 .root {
-  composes: hds-typography-body-300 from global;
+	composes: hds-typography-body-300 from global;
```
Comment (zchsh) [inline src/components/dev-dot-content/dev-dot-content.module.css:10]: Not sure what's going on with formatting here; I think this is changing spaces to tabs?

File: src/components/dev-dot-content/dev-dot-content.module.css [hunk 2]
```diff
@@ -7,93 +7,88 @@
...
-    padding-left: 14px;
-    border-top-left-radius: 2px;
-    border-bottom-left-radius: 2px;
-    border-left: 2px solid var(--token-color-border-primary);
-  }
-}
-
-.p {
-  color: var(--token-color-foreground-primary);
-  margin: 16px 0;
```
Comment (zchsh) [inline src/components/dev-dot-content/dev-dot-content.module.css:98]: Only substantial change in this file is that `.p` styles have been removed.

File: src/components/dev-dot-content/dev-dot-content.module.css (hunk 1 above)
Comment (ashleemboyer) [reply src/components/dev-dot-content/dev-dot-content.module.css:10]: How weird! That's definitely what's happening. Running prettier via the CLI command doesn't detect any issues in this file... so this must have been formatted by our lint command that runs in the pre-commit hook 🤔

**Review by ashleemboyer — APPROVED** [review]
✨

**Pull request merged**